mewarpx/mewarpx/setups_store/diode_setup.py

```diff
@@ -381,6 +383,11 @@ def init_solver(self):
                 "1D solving is not yet implemented in mewarpx")
             # self.solver = poisson1d.PoissonSolver1D()
         elif self.dim in [2, 3]:
+            if self.DIRECT_SOLVER:
+                self.solver = poisson_pseudo_1d.PoissonSolverPseudo1D(
+                    grid=self.grid
+                )
+                return
```

Comment:
I would make the code below an `else` clause rather than using `return` here. If someone happens to add more to this function below it, it seems counter-intuitive for that code to only apply to the multigrid case.